lib/pagination/adapter.js

```diff
@@ -21,10 +21,13 @@ const totalPages = adapterParams => Math.ceil(totalCount(adapterParams) / adapte
 const nextPage = adapterParams =>
   adapterParams.page >= totalPages(adapterParams) ? null : adapterParams.page + 1;
 
-const pageUrl = (adapterParams, pageParam) =>
-  pageParam
-    ? `http://${adapterParams.request.headers.host}${adapterParams.request.url}?page=${pageParam}`
-    : null;
+const pageUrl = (adapterParams, pageParam) => {
+  if (!pageParam) return null;
+
+  const urlObject = new URL(`http://${adapterParams.request.headers.host}${adapterParams.request.url}`);
```

Comment:
If we can't determine this, I see two options:
- Add a parameter with the base URL of the API and use that as a prefix
- (worst) assume https instead of http

Reply:
I like your idea of adding a parameter for the protocol or the URL. Anyway, could we cover that in this particular issue? https://github.com/Wolox/pagination-node/issues/45 . Later I can work on a PR covering specifically that situation. This PR is covering another issue with the concated pages.